File: server_pizzeria/src/app/domain/repositories/PizzaRepository.ts
import Pizza from "../entities/Pizza.js";

// Interfície per a la resposta paginada
export interface PaginatedResponse<Pizza> {
    page: number;
    per_page: number;
    total_count: number;
    records: Pizza[];
}

// Repositori abstracte
export default abstract class PizzaRepository {
    abstract getPizzes(pageNumber: number, pageSize: number): PaginatedResponse<Pizza>;
}

File: client_pizzeria/lib/domain/repository/pizza_repository.dart
// Classe de repositori: Aquesta és la part a la que l'aplicació accedeix
// per consultar les dades.

// Generalment, el repositori es divideix en dues parts:
//  * Una a la capa de domini (aquesta), que defineix una classe abstracta
//    que descriu el comportament (els mètodes), i
//  * Una altra a la capa d'infrastructura, que serà on fem la implementació
//    d'aqueseta classe, segons la representació que tinguen les dades.

import 'package:client_pizzeria/domain/entities/pizza.dart';

abstract class PizzaRepository {
  // Obté una llista de Pizzes
  Future<List<Pizza>> getPizzes({int pageNumber = 0, int pageSize = 0});
}
